linera-core/src/updater.rs

```diff
@@ -295,6 +297,28 @@ where
                     // synchronize them now and retry.
                     self.send_chain_information_for_senders(chain_id).await?;
                 }
+                Err(NodeError::EventsNotFound(event_ids)) => {
+                    let mut publisher_heights = BTreeMap::new();
+                    let new_chain_ids = event_ids
+                        .iter()
+                        .map(|event_id| event_id.chain_id)
+                        .filter(|chain_id| !publisher_chain_ids_sent.contains(chain_id))
+                        .collect::<Vec<_>>();
```

Comment:
Is it guaranteed that the `new_chain_ids` has no duplication?
If not then we could collect the `ChainId` into a `BTreeSet` and avoid repetition in the `local_node.chain_info(chain_id).await` below.